extensions/wikia/NjordPrototype/scripts/Mom.js

```diff
@@ -0,0 +1,303 @@
+				right = [];
+			$('.lcs-container .mom-module').each(function () {
+				var data = $(this).data();
+				if (typeof data.title != 'undefined') {
+					left.push(JSON.stringify({
+						'title': data.title,
+						'text': $(this).find('.mom-bar .mom-bar-content').text()
+					}));
+				}
+			});
+			$('.rcs-container .mom-module').each(function () {
+				var data = $(this).data();
+				if (typeof data.title != 'undefined') {
+					right.push(JSON.stringify({
+						'title': data.title,
+						'text': $(this).find('.mom-bar-content').text()
+					}));
+				}
+			});
+			$.nirvana.sendRequest({
+				controller: 'NjordController',
+				method: 'reorder',
+				type: 'POST',
+				data: {
+					'page': wgTitle,
+					'left': left,
+					'right': right
+				},
+				callback: onDataSaved,
+				onErrorCallback: function () {
+					// TODO: handle failure
+				}
+			});
+		}, onDataSaved = function () {
+			$mainContent.stopThrobbing();
+		}, onScroll = function () {
+			if ($(window).scrollTop() >= $mainContent.offset().top) {
+				$momHeader.addClass('mom-fixed');
+			} else {
+				$momHeader.removeClass('mom-fixed');
+			}
+		}, onDiscard = function () {
+			$editMode.hide();
+			$nonEditMode.show();
+			$mwContent.replaceWith($clonedMain);
+			$mwContent.refresh();
+			$mainContent.refresh();
+			$deleteButton.refresh();
+			$momOverlays.refresh();
+			$momBar.refresh();
+			$momBarContent.refresh();
+			$moms.refresh();
+			$leftColumn.refresh();
+			$rightColumn.refresh();
+			$deleteButton.on('click', function () {
+				$(this).parents('.mom-module').remove();
+			});
+			$('.mom-save-btn').hide();
+			$('.mom-discard-btn').hide();
+			$('.mom-wiki-markup').hide();
+		}, addEmpty = function () {
+			var $new = $(document.createElement('div')),
+				$button = $(document.createElement('div'));
+			$button.text('ADD');
+			$button.addClass('add-btn new-btn');
+			$new.addClass('mom-module mom-add-module');
+			var $secButton = $button.clone(true),
+				$secNew = $new.clone();
+			$button.on('click', {align: 'right'}, addNewModule);
+			$secButton.on('click', {align: 'left'}, addNewModule);
+
+			$rightColumn.append($new.append($button));
+			$leftColumn.append($secNew.append($secButton));
+		}, removeEmpty = function () {
+			$('.mom-add-module').remove();
+		}, addNewModule = function (ev) {
+			$mainContentContainer.startThrobbing();
+			//add placeholder will be replace after loaded
+			$('<div id="MomNewPlaceHolder" class="mom-no-display"></div>').insertBefore($(this).parent());
+			$.nirvana.sendRequest({
+				controller: 'NjordController',
+				method: 'modula',
+				format: 'HTML',
+				type: 'GET',
+				data: {
+					align: ev.data.align,
+					title: 'new'
+				},
+				callback: onAddNewBlock,
+				onErrorCallback: function () {
+					$mainContentContainer.stopThrobbing();
+					// TODO: handle failure
+				}
+			});
+		}, onAddNewBlock = function (d) {
+			var $this = $(d);
+			$('#MomNewPlaceHolder').replaceWith($this);
+			refresh();
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Move the declaration of "refresh" before this usage. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.wikia-inc.com/coding_rules#rule_key=javascript%3AVariableDeclarationAfterUsage)